docs/modules/key_did_resolver.md

```diff
@@ -0,0 +1,272 @@
+# Module: key-did-resolver
+
+## Installation
+```
+$ npm install key-did-resolver
+```
+### Usage
+This code includes support for the curves Ed25519, Secp256k1, Secp256r1 (P-256), Secp384r1 (P-384), and Secp521r1 (P-521) which follow the test vectors at:
+[https://github.com/w3c-ccg/did-method-key/tree/main/test-vectors](https://github.com/w3c-ccg/did-method-key/tree/main/test-vectors)
+
+This code has been tested with the following `did:key`[^1] providers:
+|  Curve              | Repositry                                                      |
+| ------------------- | -------------------------------------------------------------- |
+| Ed25519             | https://github.com/ceramicnetwork/key-did-provider-ed25519     |
+| Secp256k1           | https://github.com/ceramicnetwork/key-did-provider-secp256k1   |
+| P-256, P-384, P-521 | https://github.com/bshambaugh/did-key-creator                  |
+
+Compressed[^2] forms of P-256, P-384, and P-521 are preferred. [^3]
+
+[^1]: The syntax of a did:key is `did:key:id`. The did:key `id` is the `base58btc string` representation of the `Uint8Array (byte array)` consisting of the `the multicodec name` followed by `the public key {raw,uncompressed,compressed}`.
+
+[^2]: Compressed keys are the X coordinate of the public key with a prefix that depends on the sign of the Y curve coordinate. The prefix is '02' if even and '03' if odd.
+
+[^3]: During development there was not yet consensus on using all compressed keys. Support for uncompressed keys with the '04' prefix and
+raw keys (just the x,y bytes with no prefix) was kept for the P-256 and P-384 curves.
+
+### Code
+Using [@ceramicnetwork/core](https://developers.ceramic.network/reference/typescript/modules/_ceramicnetwork_core.html) with secp256k1 did-key:
+```
+import KeyDIDResolver from 'key-did-resolver'
+import {Resolver} from 'did-resolver'
+import {Ceramic} from '@ceramicnetwork/core'
+import * as IPFS from 'ipfs-core'
+import dagJose from 'dag-jose'
+import {convert} from 'blockcodec-to-ipld-format'
+
+const ipfs = await IPFS.create({
+    ipld: { formats: [dagJose] },
+})
+
+const config = {}
+const ceramic = await Ceramic.create(ipfs, config)
```

Comment:
```suggestion

```

Reply:
Looks like this file needs to be cleaned up in general. Ceramic should not be mentioned at all in here.